templates/common/page_end.html

```diff
@@ -73,6 +73,15 @@ <h2 class="typeface-a color-a">Support</h2>
 
 </div><!-- /page-container -->
 
+<script src="${CDNIFY('/static/scripts/yall.min.js')}"></script>
+<script>
+  document.addEventListener("DOMContentLoaded", function() {
```

Comment:
Rather than waiting for `DOMContentLoaded` (since this is the end of the page anyway), this can go below the `<script src="/static/scripts.js?${SHA}"></script>` and call `yall` immediately. It’ll be a little faster.

Reply:
I originally thought this wrapper could be removed, but do we actually need to wait for the thumbnail layout to do its thing? It doesn’t change the structure of the DOM from what I can tell, but it does rearrange things. If so, is `DOMContentLoaded` late enough?

On the other hand, we do want visible images to start loading as soon as possible, so maybe it’s worth calling `yall` right away even if it doesn’t have perfect information on what’s visible. `IntersectionObserver` should take care of the rest. I hope.

Reply:
This looks like it is going to be required, I think.

- When requiring that the DOMContentLoaded event fire, the page download on my test runs is around 1.34 MiB when viewing the browse and search page for submissions after a hard-refresh, and scrolling down through the page loads all the images for a total page load of 4.88 MiB for all thumbnails; so 27.45% of total file size loads just from hard-refresh.
- When calling ``yall`` directly (so ``<script>yall({threshold: 100});</script>``), the initial download from hard-refresh to idle is 4.21 MiB, with the remaining lazy elements well down in the page. This is 86.27% of the page.

I'll use a direct call if you want, but based on how my test runs are looking to get the best bang for the buck, that's going to mean waiting for ``DOMContentLoaded`` to fire, at least based on how the current page is being rendered (since a direct call seems to fire when more of the elements are on-screen, thus downloading more off-screen elements, due to the image-layout script resizing everything).

Reply:
It turns out nothing is holding space for the images, causing that problem when not waiting for thumbnail layout. Something we should probably fix later.